### ISSUE TITLE
context.getImageData fails only after image is drawn to canvas

### ISSUE THREAD

**Issue by ConorKelleher** [iddan/react-native-canvas#39]
Hey, another issue for you. Possibly related to my issue from before but, as the title says, the call to context.getImageData(...) fails after an image file has been drawn to the canvas. This can be recreated in your example project simply by adding the following line after line 41 of app.js

    context.drawImage(image, 0, 0, 100, 100); // Existing call to draw image to canvas
    context.getImageData(0, 0, 100, 100); // New async call to request the bitmapped data just drawn

By adding the above line of code, I get the following error 
    "Cannot read property 'constructor' of undefined"
And the stack trace shows the error is being thrown at Canvas.js line 169, where properties of the retrieved data are being read. 

If I put that line of code elsewhere in the file, it runs fine. For instance, if I print the value returned from awaiting that call, I can see all of the pixel data of the circle, the path, the gradient, etc. and even if I put the line of code ABOVE the call to context.drawImage(...), it returns fine with a 100x100 grid of blank pixels. It only ever seems to throw the error if I make the call after drawing an image to the canvas. I've also tried setting a timeout on the getImageData call to ensure the image has fully completed drawing.

This issue presents on all my android devices but not iOS.

I'll be happy to be of help if you need more data. Hopefully you can recreate the issue as easily as me

**Comment by ConorKelleher** [comment]
And just for extra clarity, this is tested using the latest version of your repo and the example project inside

**Comment by ConorKelleher** [comment]
By throwing a few console.logs into Canvas.js, I found that the problem is that sometimes the data variable that is returned can be an error message and this isn't being checked. The full JSON of the erroneous parsed data I got is as follows:

    {
        "id":0.022208261900120263,
        "type":"error",
        "payload":{
            "message": "Failed to execute 'getImageData' on 'CanvasRenderingContext2D': The canvas has been tainted by cross-origin data."
        }
    }

So the Canvas.js file has the issue that, while it's checking that data has a payload property, it never checks whether or not that payload property contains an error message. 

Regardless, the real issue is this whole "cross-origin data" tainting the canvas. Any idea how to remedy this?

**Comment by ConorKelleher** [comment]
One last thing, sorry about my inconsistent line numberings. I just realised I was quoting the src version of app.js and the minimised version of Canvas.js. The line in the src version of Canvas.js that throws the error is 80

**Comment by ConorKelleher** [comment]
Anything? I've looked into the issue a bit since I posted this and I really think it's a bug in the Android code. I know some images have CORS permissions in place stopping cross-origin use of images from stealing IP but I can't find a single image that works on Android, meanwhile every image I've tried has worked fine on iOS. The getImageData function works fine on iOS but on Android, it refuses to complete because of the canvas being "tainted by cross-origin data". It's preventing any progress with my release

**Comment by iddan** [comment]
Sorry, missed that will look into it ASAP

**Comment by ConorKelleher** [comment]
Were you ever able to reproduce the issue? We're looking to release our app in the next few days. iOS is fully functional but we're just delaying the android release indefinitely...

**Comment by iddan** [comment]
Haven't got to it. Pretty busy and don't use feature at work so can't find time for it in the near future

**Comment by hasaezs** [comment]
Any updates about this? A temporary fix? @ConorKelleher

**Comment by ConorKelleher** [comment]
None at all. I had to adopt a completely different approach for Android since this component was unusable for my purpose. I needed the function to get the pixel data of an image to manipulate it so I had to adapt the answer given here https://stackoverflow.com/questions/38098217/how-to-access-image-pixel-data-in-react-native to achieve that on Android. It's much slower and gives much less control over the quality of data to be read but it's a passable workaround.
